bytes/bytes_pool.go

```diff
@@ -0,0 +1,85 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package gxbytes
+
+import (
+	"errors"
+	"sync"
+)
+
+var ErrSizeTooLarge = errors.New(`size is too large`)
+
+type BytesPool struct {
+	sizes  []int // sizes declare the cap of each slot
+	slots  []sync.Pool
+	length int
+}
+
+var defaultBytesPool = NewBytesPool([]int{16, 1 << 10, 2 << 10, 4 << 10, 8 << 10, 16 << 10, 32 << 10, 64 << 10})
```

Comment:
the minimum size of bytes is one MB, which is too big. How about 16K, I think it's suitable for most cases.

Reply:
how about set the max size of bytes to 128KB?